### PR TITLE
feat(sandwich): Add optional payment step to order sandwich flow

### PR DESCRIPTION
**Pull request by lucasvanhalst**
Accept (rebase) #851 first

### CODE REVIEW

File: src/solutions/common/templates/flows/order_sandwich.xml [hunk 1]
```diff
@@ -282,6 +285,12 @@
                         nextStepResult.outlet = 'message_type';
                         nextStepResult.defaultValue = flowParams.sandwichType
                     }
+                    {% if settings.payment_enabled %}
```
Comment (RubenMattan) [inline src/solutions/common/templates/flows/order_sandwich.xml]: Payment settings are unique for every identity. So this should come from service data

File: src/solutions/common/templates/flows/order_sandwich.xml [hunk 2]
```diff
@@ -306,16 +315,55 @@
                         }
                         return canOrder;
                     }
+
+                    {% if settings.payment_enabled %}
```
Comment (RubenMattan) [inline src/solutions/common/templates/flows/order_sandwich.xml]: Payment settings are unique for every identity. So this should come from service data

File: src/solutions/common/templates/flows/order_sandwich.xml (hunk 1 above)
Comment (RubenMattan) [reply src/solutions/common/templates/flows/order_sandwich.xml]: also payment should only be shown if show_prices is True

File: src/solutions/common/templates/flows/order_sandwich.xml [hunk 3]
```diff
@@ -383,5 +431,124 @@
...
+                            return options[id];
+                        default:
+                            return 0;
+                    }
+                }
+
+                function run(rogerthat, messageFlowRun) {
+                    var amount = 0;
+
+                    for (var _i = 0, steps_1 = messageFlowRun.steps; _i &lt; steps_1.length; _i++) {
```
Comment (RubenMattan) [inline src/solutions/common/templates/flows/order_sandwich.xml]: i would not calculate the amount if payment is not enabled

**Comment by lucasvanhalst** [comment]
This will have to be updated after merging #851

**Comment by lucasvanhalst** [comment]
See https://github.com/our-city-app/oca-backend/compare/conversion_ratio...order_sandwich_pay for actual diff

**Pull request closed without merging**